### DISCUSSION TITLE
[General] bugfix in `sort_storage_access_queries` function of era-zkevm_test_harness repo

### DISCUSSION THREAD

**Discussion by peinlcy** [zkSync-Community-Hub/zksync-developers#266] [General]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

Fix a logic bug in sort and deduplicated storage logs: https://github.com/matter-labs/era-zkevm_test_harness/pull/46

There is a bug in `sort_storage_access_queries` function

```
    sorted_storage_queries_with_extra_timestamp.par_sort_by(|a, b| {
        match a.raw_query.shard_id().cmp(&a.raw_query.shard_id()) {
        ........
    }
```
It should compare the shard_id of `a` with the shard_id of `b`, the current code compares the shard_id of `a` with itself, it will result in wrong sorted_storage_queries and cause panic in the upcoming database updates.

### What is the specific issue or error you're encountering?

he impact of this bug is that if a l1_batch contains storage_logs with different shard_ids, zksync-era will fail to seal the l1_batch because of the wrong `deduplicated_storage_queries`, and in some cases, it even crashes. The issue arises when using the sort_storage_access_queries method to merge and deduplicate storage_logs, resulting in returned deduped_log_queries containing  repeated addresses and keys(due to bug). This, in turn, leads to database write failures (methods like insert_protective_reads and insert_initial_writes will crash due to primary key conflict). 

### Can you share the error messages or logs you're receiving, if any?

None

### Have you made any recent changes to the contract before encountering this issue?

None

### Are there any external libraries or contracts that your contract interacts with?

None

### Can you provide the relevant portions of your contract code where the issue is occurring?

None

### Have you tried to isolate the problem, and if so, what were the results?

If a l1_batch contains storage logs with different shard_ids, the zksync server will panic.

### What steps have you already taken to try to resolve the issue?

I have submitted a pull request to the repo to fix the bug:
https://github.com/matter-labs/era-zkevm_test_harness/pull/46

### Repo Link (Optional)

https://github.com/matter-labs/era-zkevm_test_harness

**Comment by MexicanAce** [comment]
Hi @peinlcy 👋 !

Closing in favour of your open PR, we've pinged the relevant team and apologies for the wait!
